src/core/annotations/qgsannotationitem.h

```diff
@@ -272,6 +274,66 @@ class CORE_EXPORT QgsAnnotationItem
      */
     void setSymbologyReferenceScale( double scale ) { mReferenceScale = scale; }
 
+    /**
+     * Returns the item's callout renderer, responsible for drawing item callouts.
+     *
+     * Ownership is not transferred.
+     *
+     * By default items do not have a callout, and it is necessary to be explicitly set
+     * a callout style (via setCallout() ) and set the callout anchor geometry (via set
+     * setCalloutAnchor() ).
+     *
+     * \note Callouts are only supported by items which return Qgis::AnnotationItemFlag::SupportsCallouts from flags().
+     *
+     * \see setCallout()
+     * \see calloutAnchor()
+     * \since QGIS 3.40
+     */
+    QgsCallout *callout() const;
+
+    /**
+     * Sets the item's \a callout renderer, responsible for drawing item callouts.
+     *
+     * Ownership of \a callout is transferred to the item.
+     *
+     * \note Callouts are only supported by items which return Qgis::AnnotationItemFlag::SupportsCallouts from flags().
+     *
+     * \see callout()
+     * \see setCalloutAnchor()
+     * \since QGIS 3.40
+     */
+    void setCallout( QgsCallout *callout SIP_TRANSFER );
+
+    /**
+     * Returns the callout's anchor geometry.
+     *
+     * The anchor dictates the geometry which the option item callout() should connect to. Depending on the
+     * callout subclass and anchor geometry type, the actual shape of the rendered callout may vary.
+     *
+     * The callout anchor geometry is in the parent layer's coordinate reference system.
```

Comment:
For project annotations, that would be the project CRS? Might be worth expanding here.

Reply:
It's always the layer's CRS. The main annotation layer for a project still has it's own CRS, which may or may not be the same as the project's CRS.

Reply:
@nyalldawson , fair enough.